r2/r2/lib/validator/validator.py

```diff
@@ -2736,7 +2736,7 @@ def __init__(self, item, *a, **kw):
 
     def _lookup(self, name, allow_deleted):
         try:
-            return Account._by_name(name, allow_deleted=allow_deleted)
+            return Account._by_name(name.strip(), allow_deleted=allow_deleted)
```

Comment:
What if `Account._by_name`() handled stripping and any other preconditions check for a valid username instead? That way each caller wouldn't have to worry about this and there'd be some consistency to the preconditions on `name`.